Mathlib/Algebra/Group/Pointwise/Set/Basic.lean

```diff
@@ -1328,6 +1375,16 @@ lemma preimage_mul (hm : Injective m) {s t : Set β} (hs : s ⊆ range m) (ht :
 
 end Mul
 
+section Monoid
+variable [Monoid α] [Monoid β] [FunLike F α β] [MonoidHomClass F α β]
+
+@[to_additive]
+lemma image_pow (f : F) (s : Set α) : ∀ n, f '' (s ^ n) = (f '' s) ^ n
+  | 0 => by simp [singleton_one]
+  | n + 1 => by simp [image_mul, pow_succ, image_pow]
```

Comment:
Can we also have the version of this resembling
`[MulHomClass F α β] (f : F) (s : Set α) : ∀ n, n ≠ 0 → f '' (s ^ n) = (f '' s) ^ n`?